### PR TITLE
Add 'routes:reload' task to reload Router's routes

### PR DESCRIPTION
**Pull request by ChrisBAshton**
Sometimes Router API and Router can get out of sync with one
another, e.g. immediately after a data sync. This should correct
itself as soon as a route is added, edited or removed, as Router
API then gets Router to reload all routes. However, if no route
changes, it is useful to be able to force a Router reload via
manual means.

We intend to call this Rake task post data sync in govuk-puppet.

Depended on by https://github.com/alphagov/govuk-puppet/pull/10908.

### CODE REVIEW

**Comment by ChrisBAshton** [comment]
Superseded by https://github.com/alphagov/router-api/pull/369.

File: lib/tasks/backend.rake [hunk 1]
```diff
@@ -1,4 +1,11 @@
 namespace :backend do
```
Comment (thomasleese) [inline lib/tasks/backend.rake:1]: I think `backend` in this context refers specifically to the [`Backend` model](https://github.com/alphagov/router-api/blob/master/app/models/backend.rb). I'd suggest having a `routes` namespace instead (and although probably a little much for this PR but we could also combine together [`delete_routes`](https://github.com/alphagov/router-api/blob/master/lib/tasks/delete_routes.rake) and [`dump_routes`](https://github.com/alphagov/router-api/blob/master/lib/tasks/dump_routes.rake) into it as well).

Comment (ChrisBAshton) [reply lib/tasks/backend.rake]: Fair point. I've moved the task and pushed up the commit.

[delete_routes](https://github.com/alphagov/router-api/blob/master/lib/tasks/delete_routes.rake) looks _very_... hardcoded. Does anyone run these tasks anymore? It looks to me as though we can safely delete the whole file. [dump_routes](https://github.com/alphagov/router-api/blob/master/lib/tasks/dump_routes.rake) looks a bit more generic and useful and could be moved.

Comment (thomasleese) [reply lib/tasks/backend.rake]: Yeah, I have no idea really, I think deleting that `delete_routes` file is reasonable.

Comment (ChrisBAshton) [reply lib/tasks/backend.rake]: I've found all the reassurance I need! https://github.com/alphagov/router-api/pull/95#pullrequestreview-20214326

**Review by thomasleese — APPROVED** [review]

**Pull request merged**